processing/src/main/java/org/apache/druid/segment/join/Joinables.java

```diff
@@ -114,4 +118,37 @@ public static boolean isPrefixedBy(final String columnName, final String prefix)
       );
     }).collect(Collectors.toList());
   }
+
+  public static void checkPreJoinableClausesForDuplicatesAndShadowing(
+      final List<PreJoinableClause> preJoinableClauses
+  )
+  {
+    List<String> prefixes = new ArrayList<>();
+    for (PreJoinableClause clause : preJoinableClauses) {
+      prefixes.add(clause.getPrefix());
+    }
+
+    checkPrefixesForDuplicatesAndShadowing(prefixes);
+  }
+
+  public static void checkPrefixesForDuplicatesAndShadowing(
+      final List<String> prefixes
+  )
+  {
+    for (int i = 0; i < prefixes.size(); i++) {
+      String prefix1 = prefixes.get(i);
+      for (int k = 0; k < prefixes.size(); k++) {
+        if (i != k) {
+          String otherPrefix = prefixes.get(k);
+          if (prefix1.equals(otherPrefix)) {
+            throw new IAE("Detected duplicate prefix in join clauses: [%s]", prefix1);
+          }
+
+          if (isPrefixedBy(prefix1, otherPrefix)) {
```

Comment:
this function already exists, but I think it's faster to change the order of the boolean expression in isPrefixedBy

```
return columnName.length() > prefix.length() && columnName.startsWith(prefix);
```

Reply:
cool, changed the order there